diff --git a/b/Makefile.am b/a/Makefile.am
index 4e86954..ee0dc94 100644
--- a/b/Makefile.am
+++ b/a/Makefile.am
@@ -13,7 +13,7 @@ pixmapdir       = $(datadir)/pixmaps
 xsddir          = $(datadir)/openbox
 appsdir         = $(datadir)/applications
 
-theme = Clearlooks
+theme = Adwaita-Dark
 
 AUTOMAKE_OPTIONS = subdir-objects foreign
 
@@ -351,6 +351,24 @@ clearlooks_themedir = $(themedir)/Clearlooks/openbox-3
 dist_clearlooks_theme_DATA= \
 	themes/Clearlooks/openbox-3/themerc
 
+adwaita_themedir = $(themedir)/Adwaita/openbox-3
+dist_adwaita_theme_DATA= \
+	themes/Adwaita/openbox-3/themerc \
+	themes/Adwaita/openbox-3/bullet.xbm \
+	themes/Adwaita/openbox-3/close.xbm \
+	themes/Adwaita/openbox-3/iconify.xbm \
+	themes/Adwaita/openbox-3/max.xbm \
+	themes/Adwaita/openbox-3/max_toggled.xbm
+
+adwaita_dark_themedir = $(themedir)/Adwaita-Dark/openbox-3
+dist_adwaita_dark_theme_DATA= \
+	themes/Adwaita-Dark/openbox-3/themerc \
+	themes/Adwaita-Dark/openbox-3/bullet.xbm \
+	themes/Adwaita-Dark/openbox-3/close.xbm \
+	themes/Adwaita-Dark/openbox-3/iconify.xbm \
+	themes/Adwaita-Dark/openbox-3/max.xbm \
+	themes/Adwaita-Dark/openbox-3/max_toggled.xbm
+
 ## public headers ##
 
 rrpubinclude_HEADERS = \
diff --git a/b/data/rc.xml b/a/data/rc.xml
index 3e5554b..c531efd 100644
--- a/b/data/rc.xml
+++ b/a/data/rc.xml
@@ -49,7 +49,7 @@
 </placement>
 
 <theme>
-  <name>Clearlooks</name>
+  <name>Adwaita-Dark</name>
   <titleLayout>NLIMC</titleLayout>
   <!--
       available characters are NDSLIMC, each can occur at most once.
diff --git a/a/themes/Adwaita/openbox-3/bullet.xbm b/a/themes/Adwaita/openbox-3/bullet.xbm
new file mode 100644
index 0000000..4d1b72d
--- /dev/null
+++ b/a/themes/Adwaita/openbox-3/bullet.xbm
@@ -0,0 +1,5 @@
+#define bullet_width 11
+#define bullet_height 11
+static unsigned char bullet_bits[] = {
+   0x00, 0x00, 0x00, 0x00, 0x20, 0x00, 0x60, 0x00, 0xe0, 0x00, 0xe0, 0x01,
+   0xe0, 0x00, 0x60, 0x00, 0x20, 0x00, 0x00, 0x00, 0x00, 0x00 };
diff --git a/a/themes/Adwaita/openbox-3/close.xbm b/a/themes/Adwaita/openbox-3/close.xbm
new file mode 100644
index 0000000..cc6031f
--- /dev/null
+++ b/a/themes/Adwaita/openbox-3/close.xbm
@@ -0,0 +1,5 @@
+#define close_width 12
+#define close_height 12
+static unsigned char close_bits[] = {
+   0x00, 0x00, 0x06, 0x06, 0x0e, 0x07, 0x9c, 0x03, 0xf8, 0x01, 0xf0, 0x00,
+   0xf0, 0x00, 0xf8, 0x01, 0x9c, 0x03, 0x0e, 0x07, 0x06, 0x06, 0x00, 0x00 };
diff --git a/a/themes/Adwaita/openbox-3/iconify.xbm b/a/themes/Adwaita/openbox-3/iconify.xbm
new file mode 100644
index 0000000..28610e8
--- /dev/null
+++ b/a/themes/Adwaita/openbox-3/iconify.xbm
@@ -0,0 +1,5 @@
+#define iconify_width 12
+#define iconify_height 12
+static unsigned char iconify_bits[] = {
+   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xff, 0x0f,
+   0xff, 0x0f, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 };
diff --git a/a/themes/Adwaita/openbox-3/max.xbm b/a/themes/Adwaita/openbox-3/max.xbm
new file mode 100644
index 0000000..3f3d0df
--- /dev/null
+++ b/a/themes/Adwaita/openbox-3/max.xbm
@@ -0,0 +1,5 @@
+#define max_width 12
+#define max_height 12
+static unsigned char max_bits[] = {
+   0x00, 0x00, 0xff, 0x0f, 0xff, 0x0f, 0x03, 0x0c, 0x03, 0x0c, 0x03, 0x0c,
+   0x03, 0x0c, 0x03, 0x0c, 0x03, 0x0c, 0xff, 0x0f, 0xff, 0x0f, 0x00, 0x00 };
diff --git a/a/themes/Adwaita/openbox-3/max_toggled.xbm b/a/themes/Adwaita/openbox-3/max_toggled.xbm
new file mode 100644
index 0000000..8dc06ae
--- /dev/null
+++ b/a/themes/Adwaita/openbox-3/max_toggled.xbm
@@ -0,0 +1,5 @@
+#define max_toggled_width 12
+#define max_toggled_height 12
+static unsigned char max_toggled_bits[] = {
+   0x00, 0x00, 0xfc, 0x0f, 0xfc, 0x0f, 0x0c, 0x0c, 0xff, 0x0d, 0xff, 0x0d,
+   0x83, 0x0d, 0x83, 0x0f, 0x83, 0x0f, 0xff, 0x01, 0xff, 0x01, 0x00, 0x00 };
diff --git a/a/themes/Adwaita/openbox-3/themerc b/a/themes/Adwaita/openbox-3/themerc
new file mode 100644
index 0000000..0a8b958
--- /dev/null
+++ b/a/themes/Adwaita/openbox-3/themerc
@@ -0,0 +1,193 @@
+!! MENU SETTINGS......................................
+
+!! menu title
+
+menu.title.bg: 	         flat
+menu.title.bg.color:     #cccccc
+menu.title.bg.colorTo:   #cccccc
+menu.title.text.color: 	 #5c616c
+menu.title.text.justify: center
+
+menu.border.width: 7
+menu.border.color: #eaeaea
+
+menu.overlap.x: -12
+menu.overlap.y:	 0
+
+!! menu items
+
+menu.items.bg: 	       Flat Solid
+menu.items.bg.color:   #EAEAEA
+menu.items.text.color: #5c616c
+
+menu.items.justify:             left
+menu.items.disabled.text.color: #aeaeae
+
+menu.items.active.bg: 	      Flat
+menu.items.active.bg.color:   #858a93
+menu.items.active.text.color: #ffffff
+
+!! menu separator
+
+menu.separator.width:          1
+menu.separator.padding.width:  5
+menu.separator.padding.height: 5
+menu.separator.color:          #DEDEDE
+
+!! WINDOW SETTINGS.........................
+
+!! window title
+
+window.active.title.bg:       Flat Vertical
+window.active.title.bg.color: #E7E8EB
+
+window.inactive.title.bg: 	Flat Vertical
+window.inactive.title.bg.color: #E7E8EB
+
+!! window text
+
+padding.width: 		    5
+padding.height: 	    5
+window.label.text.justify:  center
+
+window.active.label.bg:          Parentrelative
+window.active.label.bg.color:    #E7E8EB
+window.active.label.bg.colorTo:  #E7E8EB
+window.active.label.text.color:  #5c616c
+
+window.inactive.label.bg: 	   Parentrelative
+window.inactive.label.bg.color:    #e7e8eb
+window.inactive.label.bg.colorTo:  #e7e8eb
+window.inactive.label.text.color:  #9a9a9a
+
+!! window borders
+
+border.width: 1
+border.color: #ededed
+
+window.client.padding.width:  0
+window.client.padding.height: 0
+
+window.active.client.color:   #DEDEDE
+window.inactive.client.color: #DEDEDE
+
+window.active.title.separator.color:   #DEDEDE
+window.inactive.title.separator.color: #DEDEDE
+
+
+!! window handle and grip
+
+window.handle.width: 0
+
+window.active.handle.bg:       Flat Vertical
+window.active.handle.bg.color: #DEDEDE
+
+window.active.grip.bg: 	     Flat Vertical
+window.active.grip.bg.color: #DEDEDE
+
+window.inactive.handle.bg: 	 Flat Vertical
+window.inactive.handle.bg.color: #DEDEDE
+
+window.inactive.grip.bg:       Flat Vertical
+window.inactive.grip.bg.color: #DEDEDE
+
+!! WINDOW BUTTONS........................................
+
+!! ACTIVE
+
+window.active.border.color: #ededed
+
+window.active.title.bg: flat solid
+window.active.title.bg.color: #ededed
+window.active.title.separator.color: #ededed
+window.active.text.justify: center
+
+window.active.label.bg: parentrelative
+window.active.label.text.color: #5c616c
+
+window.active.handle.bg: flat solid
+window.active.handle.bg.color: #ededed
+
+window.active.grip.bg: flat solid
+window.active.grip.bg.color: #ededed
+
+window.active.button.unpressed.bg: flat solid
+window.active.button.unpressed.bg.color: #ededed
+window.active.button.unpressed.image.color: #818181
+
+window.active.button.hover.bg: flat solid
+window.active.button.hover.bg.color: #1d90cd
+window.active.button.hover.image.color: #ffffff
+
+window.active.button.pressed.bg: flat solid
+window.active.button.pressed.bg.color: #aa0000
+window.active.button.pressed.image.color: #ffffff
+
+window.active.button.disabled.bg: flat solid
+window.active.button.disabled.bg.color: #ededed
+window.active.button.disabled.image.color: #ededed
+
+window.active.button.toggled.unpressed.bg: flat solid
+window.active.button.toggled.unpressed.bg.color: #ededed
+window.active.button.toggled.unpressed.image.color: #818181
+window.active.button.toggled.hover.bg: flat solid
+window.active.button.toggled.hover.bg.color: #1d90cd
+window.active.button.toggled.hover.image.color: #ffffff
+window.active.button.toggled.pressed.bg: flat solid
+window.active.button.toggled.pressed.bg.color: #aa0000
+window.active.button.toggled.pressed.image.color: #ffffff
+
+!! INACTIVE
+
+window.inactive.border.color: #ededed
+
+window.inactive.title.bg: flat solid
+window.inactive.title.bg.color: #ededed
+window.inactive.title.separator.color: #ededed
+window.inactive.text.justify: center
+
+window.inactive.label.bg: parentrelative
+window.inactive.label.text.color: #aeaeae
+
+window.inactive.handle.bg: flat solid
+window.inactive.handle.bg.color: #ededed
+
+window.inactive.grip.bg: flat solid
+window.inactive.grip.bg.color: #ededed
+
+window.inactive.button.unpressed.bg: flat solid
+window.inactive.button.unpressed.bg.color: #ededed
+window.inactive.button.unpressed.image.color: #aeaeae
+
+window.inactive.button.hover.bg: flat solid
+window.inactive.button.hover.bg.color: #9ac8f8
+window.inactive.button.hover.image.color: #ffffff
+
+window.inactive.button.pressed.bg: flat solid
+window.inactive.button.pressed.bg.color: #f15d5d
+window.inactive.button.pressed.image.color: #ffffff
+
+window.inactive.button.disabled.bg: flat solid
+window.inactive.button.disabled.bg.color: #ededed
+window.inactive.button.disabled.image.color: #ededed
+
+window.inactive.button.toggled.unpressed.bg: flat solid
+window.inactive.button.toggled.unpressed.bg.color: #ededed
+window.inactive.button.toggled.unpressed.image.color: #aeaeae
+window.inactive.button.toggled.hover.bg: flat solid
+window.inactive.button.toggled.hover.bg.color: #9ac8f8
+window.inactive.button.toggled.hover.image.color: #ffffff
+window.inactive.button.toggled.pressed.bg: flat solid
+window.inactive.button.toggled.pressed.bg.color: #aa0000
+window.inactive.button.toggled.pressed.image.color: White
+
+!! Global width settings
+
+padding.width: 12
+padding.height: 5
+
+### Fonts
+window.active.label.text.font:shadow=n:shadowtint=10:shadowoffset=1
+window.inactive.label.text.font:shadow=n:shadowtint=10:shadowoffset=1
+menu.items.font:
+menu.title.text.font:shadow=n:shadowtint=0
diff --git a/a/themes/Adwaita-Dark/openbox-3/bullet.xbm b/a/themes/Adwaita-Dark/openbox-3/bullet.xbm
new file mode 100644
index 0000000..4d1b72d
--- /dev/null
+++ b/a/themes/Adwaita-Dark/openbox-3/bullet.xbm
@@ -0,0 +1,5 @@
+#define bullet_width 11
+#define bullet_height 11
+static unsigned char bullet_bits[] = {
+   0x00, 0x00, 0x00, 0x00, 0x20, 0x00, 0x60, 0x00, 0xe0, 0x00, 0xe0, 0x01,
+   0xe0, 0x00, 0x60, 0x00, 0x20, 0x00, 0x00, 0x00, 0x00, 0x00 };
diff --git a/a/themes/Adwaita-Dark/openbox-3/close.xbm b/a/themes/Adwaita-Dark/openbox-3/close.xbm
new file mode 100644
index 0000000..cc6031f
--- /dev/null
+++ b/a/themes/Adwaita-Dark/openbox-3/close.xbm
@@ -0,0 +1,5 @@
+#define close_width 12
+#define close_height 12
+static unsigned char close_bits[] = {
+   0x00, 0x00, 0x06, 0x06, 0x0e, 0x07, 0x9c, 0x03, 0xf8, 0x01, 0xf0, 0x00,
+   0xf0, 0x00, 0xf8, 0x01, 0x9c, 0x03, 0x0e, 0x07, 0x06, 0x06, 0x00, 0x00 };
diff --git a/a/themes/Adwaita-Dark/openbox-3/iconify.xbm b/a/themes/Adwaita-Dark/openbox-3/iconify.xbm
new file mode 100644
index 0000000..28610e8
--- /dev/null
+++ b/a/themes/Adwaita-Dark/openbox-3/iconify.xbm
@@ -0,0 +1,5 @@
+#define iconify_width 12
+#define iconify_height 12
+static unsigned char iconify_bits[] = {
+   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xff, 0x0f,
+   0xff, 0x0f, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 };
diff --git a/a/themes/Adwaita-Dark/openbox-3/max.xbm b/a/themes/Adwaita-Dark/openbox-3/max.xbm
new file mode 100644
index 0000000..3f3d0df
--- /dev/null
+++ b/a/themes/Adwaita-Dark/openbox-3/max.xbm
@@ -0,0 +1,5 @@
+#define max_width 12
+#define max_height 12
+static unsigned char max_bits[] = {
+   0x00, 0x00, 0xff, 0x0f, 0xff, 0x0f, 0x03, 0x0c, 0x03, 0x0c, 0x03, 0x0c,
+   0x03, 0x0c, 0x03, 0x0c, 0x03, 0x0c, 0xff, 0x0f, 0xff, 0x0f, 0x00, 0x00 };
diff --git a/a/themes/Adwaita-Dark/openbox-3/max_toggled.xbm b/a/themes/Adwaita-Dark/openbox-3/max_toggled.xbm
new file mode 100644
index 0000000..8dc06ae
--- /dev/null
+++ b/a/themes/Adwaita-Dark/openbox-3/max_toggled.xbm
@@ -0,0 +1,5 @@
+#define max_toggled_width 12
+#define max_toggled_height 12
+static unsigned char max_toggled_bits[] = {
+   0x00, 0x00, 0xfc, 0x0f, 0xfc, 0x0f, 0x0c, 0x0c, 0xff, 0x0d, 0xff, 0x0d,
+   0x83, 0x0d, 0x83, 0x0f, 0x83, 0x0f, 0xff, 0x01, 0xff, 0x01, 0x00, 0x00 };
diff --git a/a/themes/Adwaita-Dark/openbox-3/themerc b/a/themes/Adwaita-Dark/openbox-3/themerc
new file mode 100644
index 0000000..c7e97ad
--- /dev/null
+++ b/a/themes/Adwaita-Dark/openbox-3/themerc
@@ -0,0 +1,181 @@
+!! MENU SETTINGS......................................
+
+!! Menu background
+menu.items.bg: flat gradient vertical
+menu.items.bg.color: #33393B
+menu.items.bg.colorTo: #33393B
+menu.border.width: 7
+menu.border.color: #33393B
+menu.overlap.x: -12
+menu.overlap.y:	0
+
+!! Menu text
+menu.items.text.color: #A4A5A9
+menu.items.justify: left
+menu.items.disabled.text.color: #606060
+
+!! Menu headers
+menu.title.bg: flat raised splitvertical gradient
+menu.title.bg.color: #33393B
+menu.title.bg.colorTo: #33393B
+menu.title.text.color: #A4A5A9
+menu.title.text.justify: center
+
+!! Selected menu item
+menu.items.active.bg: flat
+menu.items.active.bg.color: #4084D6
+menu.items.active.bg.colorTo: #4084D6
+menu.items.active.text.color: white
+
+!! WINDOW SETTINGS.........................
+
+!! window title
+
+window.active.title.bg:       Flat Vertical
+window.active.title.bg.color: #33393B
+
+window.inactive.title.bg: 	Flat Vertical
+window.inactive.title.bg.color: #33393B
+
+!! window text
+
+padding.width: 		    5
+padding.height: 	    5
+window.label.text.justify:  center
+
+window.active.label.bg:          Parentrelative
+window.active.label.bg.color:    #33393B
+window.active.label.bg.colorTo:  #33393B
+window.active.label.text.color:  #5c616c
+
+window.inactive.label.bg: 	   Parentrelative
+window.inactive.label.bg.color:    #505050
+window.inactive.label.bg.colorTo:  #505050
+window.inactive.label.text.color:  #404040
+
+!! window borders
+
+border.width: 1
+border.color: #ededed
+
+window.client.padding.width:  0
+window.client.padding.height: 0
+
+window.active.client.color:   #DEDEDE
+window.inactive.client.color: #DEDEDE
+
+window.active.title.separator.color:   #DEDEDE
+window.inactive.title.separator.color: #DEDEDE
+
+
+!! window handle and grip
+
+window.handle.width: 0
+
+window.active.handle.bg:       Flat Vertical
+window.active.handle.bg.color: #dedede
+
+window.active.grip.bg: 	     Flat Vertical
+window.active.grip.bg.color: #DEDEDE
+
+window.inactive.handle.bg: 	 Flat Vertical
+window.inactive.handle.bg.color: #DEDEDE
+
+window.inactive.grip.bg:       Flat Vertical
+window.inactive.grip.bg.color: #DEDEDE
+
+!! WINDOW BUTTONS........................................
+
+!! ACTIVE
+
+window.active.border.color: #33393b
+
+window.active.title.bg: flat solid
+window.active.title.bg.color: #33393B
+window.active.title.separator.color: #33393B
+window.active.text.justify: center
+
+window.active.label.bg: parentrelative
+window.active.label.text.color: #A4A5A9
+
+window.active.handle.bg: flat solid
+window.active.handle.bg.color: #33393B
+
+window.active.grip.bg: flat solid
+window.active.grip.bg.color: #33393B
+
+window.active.button.unpressed.bg: flat solid
+window.active.button.unpressed.bg.color: #33393B
+window.active.button.unpressed.image.color: #94969A
+
+window.active.button.hover.bg: flat solid
+window.active.button.hover.bg.color: #1d90cd
+window.active.button.hover.image.color: #ffffff
+
+window.active.button.pressed.bg: flat solid
+window.active.button.pressed.bg.color: #aa0000
+window.active.button.pressed.image.color: #ffffff
+
+window.active.button.disabled.bg: flat solid
+window.active.button.disabled.bg.color: #33393B
+window.active.button.disabled.image.color: #33393B
+
+window.active.button.toggled.unpressed.bg: flat solid
+window.active.button.toggled.unpressed.bg.color: #33393b
+window.active.button.toggled.unpressed.image.color: #94969a
+window.active.button.toggled.hover.bg: flat solid
+window.active.button.toggled.hover.bg.color: #1d90cd
+window.active.button.toggled.hover.image.color: #ffffff
+window.active.button.toggled.pressed.bg: flat solid
+window.active.button.toggled.pressed.bg.color: #aa0000
+window.active.button.toggled.pressed.image.color: #ffffff
+
+!! INACTIVE
+
+window.inactive.border.color: #33393B
+
+window.inactive.title.bg: flat solid
+window.inactive.title.bg.color: #33393B
+window.inactive.title.separator.color: #33393B
+window.inactive.text.justify: center
+
+window.inactive.label.bg: parentrelative
+window.inactive.label.text.color: #757981
+
+window.inactive.handle.bg: flat solid
+window.inactive.handle.bg.color: #2F343F
+
+window.inactive.grip.bg: flat solid
+window.inactive.grip.bg.color: #2F343F
+
+window.inactive.button.unpressed.bg: flat solid
+window.inactive.button.unpressed.bg.color: #33393B
+window.inactive.button.unpressed.image.color: #6B6B6B
+
+window.inactive.button.hover.bg: flat solid
+window.inactive.button.hover.bg.color: #79a6d6
+window.inactive.button.hover.image.color: #ffffff
+
+window.inactive.button.pressed.bg: flat solid
+window.inactive.button.pressed.bg.color: #33393B
+window.inactive.button.pressed.image.color: #3E84CE
+
+window.inactive.button.disabled.bg: flat solid
+window.inactive.button.disabled.bg.color: #33393B
+window.inactive.button.disabled.image.color: #33393b
+
+window.inactive.button.toggled.unpressed.bg: flat solid
+window.inactive.button.toggled.unpressed.bg.color: #33393b
+window.inactive.button.toggled.unpressed.image.color: #6B6B6B
+window.inactive.button.toggled.hover.bg: flat solid
+window.inactive.button.toggled.hover.bg.color: #79A6D6
+window.inactive.button.toggled.hover.image.color: #ffffff
+window.inactive.button.toggled.pressed.bg: flat solid
+window.inactive.button.toggled.pressed.bg.color: #aa0000
+window.inactive.button.toggled.pressed.image.color: #ffffff
+
+
+!! Global width settings
+
+padding.width: 12
+padding.height: 5
